rpcs3/util/atomic.cpp

```diff
@@ -27,6 +31,113 @@ static inline bool ptr_cmp(const void* data, std::size_t size, u64 old_value)
 	return false;
 }
 
+// Fallback implementation
+namespace
+{
+	struct waiter
+	{
+		std::condition_variable cond;
+		void* const tls_ptr;
+
+		explicit waiter(void* tls_ptr)
+			: tls_ptr(tls_ptr)
+		{
+		}
+	};
+
+	struct waiter_map
+	{
+		std::mutex mutex;
+		std::multimap<const void*, waiter> list;
+	};
+
+	// Thread's unique node to insert without allocation
+	thread_local std::multimap<const void*, waiter>::node_type s_tls_waiter = []()
+	{
+		// Initialize node from a dummy container (there is no separate node constructor)
+		std::multimap<const void*, waiter> dummy;
+		return dummy.extract(dummy.emplace(nullptr, &s_tls_waiter));
```

Comment:
May be a long shot, but won't constructing via an initializer list work? Or would that be relying on an implementation detail?

Reply:
There is no usable autonomous constructors for nodes in standard (except default and move).